src/main/java/view/UserPageImpl.java

```diff
@@ -1,52 +1,171 @@
         return blindHorizontal;
     }
 
+    private BlindVertical requestDataVerticalBlind() {
+        BlindVertical blindVertical = new BlindVertical();
+        System.out.println("Enter the width of the vertical blind in mm: ");
+        int width;
+        while (true) {
+            String line = scanner.nextLine();
+            if (!line.matches("^\\d{3,4}$") || Integer.parseInt(line) < 400 || Integer.parseInt(line) > 6000) {
+                System.out.println("You entered incorrect data, please enter again!\nWidth should be from 400 to 6000 mm.");
+            } else {
+                width = Integer.parseInt(line);
+                blindVertical.setWidth(width);
+                break;
+            }
+        }
+        System.out.println("Enter the height of the vertical blind in mm:");
+        int height;
+        while (true) {
+            String line = scanner.nextLine();
+            if (!line.matches("^\\d{3,4}$") || Integer.parseInt(line) < 200 || Integer.parseInt(line) > 4000) {
+                System.out.println("You entered incorrect data, please enter again!\nHeight should be from 200 to 4000 mm.");
+            } else {
+                height = Integer.parseInt(line);
+                blindVertical.setHeight(height);
+                break;
+            }
+        }
+        System.out.println("Enter the type of the vertical blind (01, 02, 03): ");
+        int type;
+        while (true) {
+            String line = scanner.nextLine();
+            if (!line.matches("^\\d{1,2}$") || Integer.parseInt(line) < 1 || Integer.parseInt(line) > 3) {
+                System.out.println("You entered incorrect data, please enter again!");
+            } else {
+                type = Integer.parseInt(line);
+                blindVertical.setType(type);
+                break;
+            }
+        }
+        System.out.println("Enter the color of the vertical blind (1(white), 2(green), 3(yellow), 4(blue), 5(beige)): ");
+        Color[] colors = Color.values();
+        int color;
+        while (true) {
+            String line = scanner.nextLine();
+            if (!line.matches("^\\d$") || Integer.parseInt(line) < 1 || Integer.parseInt(line) > 5) {
+                System.out.println("You entered incorrect data, please enter again!");
+            } else {
+                color = Integer.parseInt(line) - 1;
+                blindVertical.setColor(colors[color]);
+                break;
+            }
+        }
+        System.out.println("Enter the mount type of the vertical blind (1(ceiling), 2(wall)): ");
+        MountType[] mountTypes = MountType.values();
+        int mount;
+        while (true) {
+            String line = scanner.nextLine();
+            if (!line.matches("^\\d$") || Integer.parseInt(line) < 1 || Integer.parseInt(line) > 2) {
+                System.out.println("You entered incorrect data, please enter again!");
+            } else {
+                mount = Integer.parseInt(line) - 1;
```

Comment:
Logic looks unreliable. Let's add a new field value to MountType enum and we can get MountType by this value

Reply:
done